packages/clerk-js/src/ui/contexts/utils.ts

```diff
@@ -29,3 +29,11 @@ export function getInitialValuesFromQueryParams(queryString: string, params: str
 }
 
 export const populateParamFromObject = createDynamicParamParser({ regex: /:(\w+)/ });
+
+type Mode = 'modal' | 'mounted' | undefined;
+/**
+ * When we're in modal mode and no redirect_url is provided, we want to redirect to the current page after sign-in or sign-up.
+ */
+export const determineRedirectUrlFromMode = ({ mode, url, clerk }: { mode: Mode; url: string; clerk: Clerk }) => {
+  return mode === 'modal' && url === '/' ? window.location.href : clerk.buildUrlWithAuth(url);
```

Comment:
what if `afterSignInUrl` was set explicitly to `/` ?

Maybe this logic should exists inside `new RedirectUrls()`

Reply:
Yeah that is a good call, opened a separate PR with that approach here https://github.com/clerk/javascript/pull/4768